Sources/Commands/PackageTools/SwiftPackageTool.swift

```diff
@@ -125,15 +125,36 @@ extension SwiftPackageTool {
             else if matchingPlugins.count > 1 {
                 throw ValidationError("\(matchingPlugins.count) plugins found for '\(command)'")
             }
+
+            // handle plugin execution arguments that got passed after the plugin name
+            let pluginArguments = try PluginArguments.parse(Array(self.remaining.drop(while: { $0 == command })))
+            // merge the relevant plugin execution options
+            var pluginOptions = self._pluginOptions
+            pluginOptions.allowWritingToPackageDirectory = pluginOptions.allowWritingToPackageDirectory || pluginArguments.pluginOptions.allowWritingToPackageDirectory
+            pluginOptions.additionalAllowedWritableDirectories.append(contentsOf: pluginArguments.pluginOptions.additionalAllowedWritableDirectories)
+            swiftTool.shouldDisableSandbox = swiftTool.shouldDisableSandbox || pluginArguments.globalOptions.security.shouldDisableSandbox
```

Comment:
The issue I see with this is that it will be very easy to add an option to `PluginCommand.PluginOptions` and not add the handling here. Do we have any way to prevent that?

Reply:
yes this is the tricky part. there is not a great way afaik, we should limit the amount of argument that are relevant in this context. @natecook1000 ideas?

Reply:
You could move this into a method on `PluginOptions`, with an assert on the number of properties:

```swift
extension PluginOptions {
    func merged(with other: PluginOptions) -> PluginOptions {
        // create result
        assert(Mirror(reflecting: self).children.count == 4, "Property added to PluginOptions without updating merged(with:)!")
    }
}
```